mat/cholesky.go

```diff
@@ -664,13 +663,13 @@ func (c *Cholesky) SymRankOne(orig *Cholesky, alpha float64, x Vector) (ok bool)
 			sin[i] *= -1
 		}
 	}
-	umat := c.chol.mat
-	stride := umat.Stride
+	workMat := NewTriDense(c.chol.mat.N, c.chol.triKind(), nil)
```

Comment:
I've just noticed that there is a `getWorkspaceTri`/`putWorkspaceTri` function pair that will help avoid the allocation here.